### PR TITLE
feat: block processing implement process_randao

### LINKED ISSUE #425: [tracking] Block processing implementations
- [ ] `process_block_header`
- [ ] `process_randao`
- [ ] `process_eth1_data`
- [ ] `process_operations`

### LINKED ISSUE #437: [block processing] `process_randao`
@f3r10

### PR DESCRIPTION
**Pull request by f3r10**
**Motivation**

block processing implement process_randao

Closes #437

### CODE REVIEW

File: lib/lambda_ethereum_consensus/state_transition/operations.ex [hunk 1]
```diff
@@ -630,6 +631,50 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
...
+            mix
+          )
+
+        {:ok,
+         %BeaconState{
+           state
+           | randao_mixes: updated_randao_mixes
+         }}
+      else
+        {:error, "unable to verify public signature while executing process_randao"}
```
Comment (MegaRedHand) [inline lib/lambda_ethereum_consensus/state_transition/operations.ex:673]: Nit: the error message is too long. I think a better one would be just `"invalid randao reveal"`. wdyt?

File: lib/lambda_ethereum_consensus/state_transition/operations.ex [hunk 2]
```diff
@@ -630,6 +631,50 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
...
+      signing_root = Misc.compute_signing_root(epoch, SszTypes.Epoch, domain)
+
+      if Bls.valid?(proposer.pubkey, signing_root, randao_reveal) do
+        randao_mix = Accessors.get_randao_mix(state, epoch)
+        hash = :crypto.hash(:sha256, randao_reveal)
+
+        # Mix in RANDAO reveal
+        mix =
+          Enum.zip(:binary.bin_to_list(randao_mix), :binary.bin_to_list(hash))
+          |> Enum.map_join(fn {left, rigth} -> Bitwise.bxor(left, rigth) end)
```
Comment (MegaRedHand) [inline lib/lambda_ethereum_consensus/state_transition/operations.ex:658]: [`:crypto.exor`](https://www.erlang.org/doc/man/crypto#exor-2) can be used for this:
```suggestion
        mix = :crypto.exor(randao_mix, hash)
```

File: lib/lambda_ethereum_consensus/state_transition/operations.ex [hunk 3]
```diff
@@ -630,6 +631,50 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
...
+
+        updated_randao_mixes =
+          List.replace_at(
+            state.randao_mixes,
+            Integer.mod(epoch, ChainSpec.get("EPOCHS_PER_HISTORICAL_VECTOR")),
```
Comment (MegaRedHand) [inline lib/lambda_ethereum_consensus/state_transition/operations.ex:663]: Nit: we usually use `rem` instead of `Integer.mod` (both are equal in this case)

**Review by MegaRedHand — COMMENTED** [review]
LGTM. We can leave the other functions (from #425) for following PRs

File: lib/lambda_ethereum_consensus/state_transition/operations.ex (hunk 1 above)
Comment (f3r10) [reply lib/lambda_ethereum_consensus/state_transition/operations.ex:673]: yes, you are right :+1:

**Review by MegaRedHand — APPROVED** [review]
Good stuff! 🔥

**Pull request merged**